### PR TITLE
[multistage] Adding rule to skip SINGLE_VALUE aggregation function

### PR DESCRIPTION
**Pull request by xiangfu0**
`SINGLE_VALUE` is a data integrity check, so we can skip it.

Attached the current exceptions/stacktrace for query:
`select  sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > (select avg(ActualElapsedTime) as avg_profit from mytable)`

```
Caught exception while compiling SQL request 95713964000000001: select  sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > 
(select avg(ActualElapsedTime) as avg_profit from mytable) 
, Error composing query plan for: select  sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > 
(select avg(ActualElapsedTime) as avg_profit from mytable) 

org.apache.pinot.query.QueryEnvironment.planQuery(QueryEnvironment.java:182)
org.apache.pinot.broker.requesthandler.MultiStageBrokerRequestHandler.handleRequest(MultiStageBrokerRequestHandler.java:152)
org.apache.pinot.broker.requesthandler.BaseBrokerRequestHandler.handleRequest(BaseBrokerRequestHandler.java:263)
org.apache.pinot.broker.requesthandler.BrokerRequestHandler.handleRequest(BrokerRequestHandler.java:48)
Cannot generate a valid execution plan for the given query: LogicalAggregate(group=[{}], EXPR$0=[SUM($0)])
  LogicalProject(ActualElapsedTime=[$0])
    LogicalFilter(condition=[>($0, $1)])
      LogicalJoin(condition=[true], joinType=[left])
        LogicalProject(ActualElapsedTime=[$3])
          LogicalTableScan(table=[[mytable]])
        LogicalAggregate(group=[{}], agg#0=[SINGLE_VALUE($0)])
          LogicalAggregate(group=[{}], avg_profit=[AVG($0)])
            LogicalProject(ActualElapsedTime=[$3])
              LogicalTableScan(table=[[mytable]])

org.apache.pinot.query.QueryEnvironment.optimize(QueryEnvironment.java:342)
org.apache.pinot.query.QueryEnvironment.compileQuery(QueryEnvironment.java:284)
org.apache.pinot.query.QueryEnvironment.planQuery(QueryEnvironment.java:173)
org.apache.pinot.broker.requesthandler.MultiStageBrokerRequestHandler.handleRequest(MultiStageBrokerRequestHandler.java:152)
Invalid aggregation function name: SINGLE_VALUE
org.apache.pinot.segment.spi.AggregationFunctionType.getAggregationFunctionType(AggregationFunctionType.java:454)
org.apache.calcite.rel.rules.PinotAggregateExchangeNodeInsertRule.buildAggregateCall(PinotAggregateExchangeNodeInsertRule.java:295)
org.apache.calcite.rel.rules.PinotAggregateExchangeNodeInsertRule.convertAggForLeafInput(PinotAggregateExchangeNodeInsertRule.java:243)
org.apache.calcite.rel.rules.PinotAggregateExchangeNodeInsertRule.createPlanWithLeafExchangeFinalAggregate(PinotAggregateExchangeNodeInsertRule.java:147)

jsonNode = {"requestId":"95713964000000001","brokerId":"Broker_192.168.86.250_18099","exceptions":[{"message":"SQLParsingError:
Error composing query plan for: select  sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > 
(select avg(ActualElapsedTime) as avg_profit from mytable) 

org.apache.pinot.query.QueryEnvironment.planQuery(QueryEnvironment.java:182)
org.apache.pinot.broker.requesthandler.MultiStageBrokerRequestHandler.handleRequest(MultiStageBrokerRequestHandler.java:152)
org.apache.pinot.broker.requesthandler.BaseBrokerRequestHandler.handleRequest(BaseBrokerRequestHandler.java:263)
org.apache.pinot.broker.requesthandler.BrokerRequestHandler.handleRequest(BrokerRequestHandler.java:48)
Cannot generate a valid execution plan for the given query: LogicalAggregate(group=[{}], EXPR$0=[SUM($0)])
  LogicalProject(ActualElapsedTime=[$0])
    LogicalFilter(condition=[>($0, $1)])
      LogicalJoin(condition=[true], joinType=[left])
        LogicalProject(ActualElapsedTime=[$3])
          LogicalTableScan(table=[[mytable]])
        LogicalAggregate(group=[{}], agg#0=[SINGLE_VALUE($0)])
          LogicalAggregate(group=[{}], avg_profit=[AVG($0)])
            LogicalProject(ActualElapsedTime=[$3])
              LogicalTableScan(table=[[mytable]])

org.apache.pinot.query.QueryEnvironment.optimize(QueryEnvironment.java:342)
org.apache.pinot.query.QueryEnvironment.compileQuery(QueryEnvironment.java:284)
org.apache.pinot.query.QueryEnvironment.planQuery(QueryEnvironment.java:173)
org.apache.pinot.broker.requesthandler.MultiStageBrokerRequestHandler.handleRequest(MultiStageBrokerRequestHandler.java:152)
Invalid aggregation function name: SINGLE_VALUE
org.apache.pinot.segment.spi.AggregationFunctionType.getAggregationFunctionType(AggregationFunctionType.java:454)
org.apache.calcite.rel.rules.PinotAggregateExchangeNodeInsertRule.buildAggregateCall(PinotAggregateExchangeNodeInsertRule.java:295)
org.apache.calcite.rel.rules.PinotAggregateExchangeNodeInsertRule.convertAggForLeafInput(PinotAggregateExchangeNodeInsertRule.java:243)
org.apache.calcite.rel.rules.PinotAggregateExchangeNodeInsertRule.createPlanWithLeafExchangeFinalAggregate(PinotAggregateExchangeNodeInsertRule.java:147)
","errorCode":150}],"numServersQueried":0,"numServersResponded":0,"numSegmentsQueried":0,"numSegmentsProcessed":0,"numSegmentsMatched":0,"numConsumingSegmentsQueried":0,"numConsumingSegmentsProcessed":0,"numConsumingSegmentsMatched":0,"numDocsScanned":0,"numEntriesScannedInFilter":0,"numEntriesScannedPostFilter":0,"numGroupsLimitReached":false,"totalDocs":0,"timeUsedMs":0,"offlineThreadCpuTimeNs":0,"realtimeThreadCpuTimeNs":0,"offlineSystemActivitiesCpuTimeNs":0,"realtimeSystemActivitiesCpuTimeNs":0,"offlineResponseSerializationCpuTimeNs":0,"realtimeResponseSerializationCpuTimeNs":0,"offlineTotalCpuTimeNs":0,"realtimeTotalCpuTimeNs":0,"brokerReduceTimeMs":0,"segmentStatistics":[],"traceInfo":{},"numRowsResultSet":0,"minConsumingFreshnessTimeMs":0,"numSegmentsPrunedByBroker":0,"numSegmentsPrunedByServer":0,"numSegmentsPrunedInvalid":0,"numSegmentsPrunedByLimit":0,"numSegmentsPrunedByValue":0,"explainPlanNumEmptyFilterSegments":0,"explainPlanNumMatchAllFilterSegments":0}
```

### CODE REVIEW

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java [hunk 1]
```diff
@@ -101,6 +101,23 @@ public void testHardcodedQueries()
     super.testHardcodedQueries();
   }
 
+  @Test
+  public void testSingleValueQuery()
+      throws Exception {
+    String query = "select  sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > \n"
+        + "(select avg(ActualElapsedTime) as avg_profit from mytable) \n";
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java:108]:
```suggestion
    String query = "select sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > "
        + "(select avg(ActualElapsedTime) as avg_profit from mytable)";
```

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java [hunk 2]
```diff
@@ -101,6 +101,23 @@ public void testHardcodedQueries()
...
+    JsonNode jsonNode = postQuery(query);
+    long joinResult = jsonNode.get("resultTable").get("rows").get(0).get(0).asLong();
+    System.out.println("jsonNode = " + jsonNode);
+
+    // The query of `SELECT avg(ActualElapsedTime) FROM mytable` is -1412.435033969449
+    query = "select  sum(ActualElapsedTime) as profit from mytable WHERE ActualElapsedTime > -1412.435033969449 \n";
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java:114]:
```suggestion
    query = "select  sum(ActualElapsedTime) as profit from mytable WHERE ActualElapsedTime > -1412.435033969449";
```

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java [hunk 3]
```diff
@@ -101,6 +101,23 @@ public void testHardcodedQueries()
...
   }
 
+  @Test
+  public void testSingleValueQuery()
+      throws Exception {
+    String query = "select  sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > \n"
+        + "(select avg(ActualElapsedTime) as avg_profit from mytable) \n";
+    JsonNode jsonNode = postQuery(query);
+    long joinResult = jsonNode.get("resultTable").get("rows").get(0).get(0).asLong();
+    System.out.println("jsonNode = " + jsonNode);
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java:111]: Remove

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java [hunk 4]
```diff
@@ -101,6 +101,23 @@ public void testHardcodedQueries()
...
+
+    // The query of `SELECT avg(ActualElapsedTime) FROM mytable` is -1412.435033969449
+    query = "select  sum(ActualElapsedTime) as profit from mytable WHERE ActualElapsedTime > -1412.435033969449 \n";
+    jsonNode = postQuery(query);
+    System.out.println("jsonNode = " + jsonNode);
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java:116]: Remove

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java [hunk 5]
```diff
@@ -101,6 +101,23 @@ public void testHardcodedQueries()
     super.testHardcodedQueries();
   }
 
+  @Test
+  public void testSingleValueQuery()
+      throws Exception {
+    String query = "select  sum(ActualElapsedTime) from mytable WHERE ActualElapsedTime > \n"
+        + "(select avg(ActualElapsedTime) as avg_profit from mytable) \n";
+    JsonNode jsonNode = postQuery(query);
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java:109]: Can we compare with H2 result?

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java (hunk 5 above)
Comment (xiangfu0) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java:109]: No, 
Integration test result comparison needs to parse the query using v1 parser, which throws exceptions.
```
Caused by: java.lang.ClassCastException: class org.apache.calcite.sql.SqlSelect cannot be cast to class org.apache.calcite.sql.SqlBasicCall (org.apache.calcite.sql.SqlSelect and org.apache.calcite.sql.SqlBasicCall are in unnamed module of loader 'app')
	at org.apache.pinot.sql.parsers.CalciteSqlParser.toExpression(CalciteSqlParser.java:748)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.compileFunctionExpression(CalciteSqlParser.java:810)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.toExpression(CalciteSqlParser.java:748)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.compileSqlNodeToPinotQuery(CalciteSqlParser.java:458)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.compileToPinotQuery(CalciteSqlParser.java:191)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.compileToPinotQuery(CalciteSqlParser.java:186)
	at org.apache.pinot.core.query.request.context.utils.QueryContextConverterUtils.getQueryContext(QueryContextConverterUtils.java:48)
	at org.apache.pinot.integration.tests.ClusterIntegrationTestUtils.testQueryInternal(ClusterIntegrationTestUtils.java:729)
	at org.apache.pinot.integration.tests.ClusterIntegrationTestUtils.testQuery(ClusterIntegrationTestUtils.java:677)
	... 27 more
```

**Pull request merged**